common/types/shared.go

```diff
@@ -5572,6 +5689,76 @@ type SignalWithStartWorkflowExecutionRequest struct {
 	JitterStartSeconds                  *int32                 `json:"jitterStartSeconds,omitempty"`
 }
 
+func (v *SignalWithStartWorkflowExecutionRequest) SerializeForLogging() (string, error) {
```

Comment:
As a broad bit of critique / feedback:
Fairly often this kind of thing gets missed / falls out of sync during future changes to a structure (though I admit many of these structs are _relatively_ unlikely to see changes that would be a significant problem).

A less-fragile option would be to use reflection and struct tags, as those are immediately visible when you make any modification to the struct.  Then all of these (a shallow + shallow-copy-slice-contents copy) could be built once and not need to be customized.

---

Otherwise, as this is not a more standard `x = y` copy or a deep-copy like "`json.Marshal` -> `json.Unmarshal`": why are values being selected / ignored, and why do semi-deep copies at all?

I don't mean to imply there is no reason or this should not be done, but diverging from a common operation should be justified, as it will be surprising to people in the future and needs to be maintained.
- Are we worried about mutation, and this non-mutated copy is more correct somehow?
- Would it be reasonable to do a shallow copy and then erase specific fields?  e.g. not logging `Input` seems completely reasonable, but `dup = *v; dup.Input = nil` is trivial compared to this code.
- Related to struct tags: much of this could be accomplished by `Input []byte 'json:"-"'` as this is immediately json-marshaled.  Would that be better?  Would it cause issues in some other logging / json-ifying location?
- Zap (the underlying logger) has a MarshalLogObject API pretty much entirely for this purpose.  Can we use it / do we have a mirror?  And can we generate it instead of writing it by hand?

Reply:
After a great discussion, decided to use json tags to filter PII.

Reply:
for more context: we don't rely on these json tags for any important _behavior_ as far as we are aware (the codebase is quite consistent about explicitly mapping between types, not monkey-patching with serialization), and we can't find any logs including even "input" so it's quite likely these are all unused.